src/components/tx-flow/common/TxNonce/index.tsx

```diff
@@ -104,12 +104,20 @@ const TxNonceForm = ({ nonce, recommendedNonce }: { nonce: string; recommendedNo
     defaultValues: {
       [TxNonceFormFieldNames.NONCE]: nonce,
     },
-    mode: 'onTouched',
+    mode: 'all',
     values: {
       [TxNonceFormFieldNames.NONCE]: nonce,
     },
   })
 
+  // Update the nonce in the form if the user hasn't changed the value manually
+  // and the recommendedNonce updated in the background
+  useEffect(() => {
+    if (formMethods.formState.touchedFields[TxNonceFormFieldNames.NONCE]) return
+
+    formMethods.setValue(TxNonceFormFieldNames.NONCE, recommendedNonce)
```

Comment:
Why is needed? Can’t the input just be a controlled input?

Reply:
Good point, I think the issue is that the `validate` calls `setNonce` automatically without user input so that the controlled input value `nonce` will never update to the recommended nonce. I've added an early return inside the `validate` function to skip validation if the input value is the same as `nonce` to avoid initial validation. This fixes the issue too and since we know that `nonce` is always valid it is ok to skip validation if the input value is the same.